datafusion/expr/src/built_in_function.rs

```diff
@@ -1392,8 +1399,8 @@ macro_rules! make_utf8_to_return_type {
     ($FUNC:ident, $largeUtf8Type:expr, $utf8Type:expr) => {
         fn $FUNC(arg_type: &DataType, name: &str) -> Result<DataType> {
             Ok(match arg_type {
-                DataType::LargeUtf8 => $largeUtf8Type,
-                DataType::Utf8 => $utf8Type,
+                DataType::LargeUtf8 | DataType::LargeBinary=> $largeUtf8Type,
```

Comment:
I am not sure about this -- this implies that any function that takes a binary input will produce binary output which doesn't seem right

Reply:
I take it back -- this change works. I have incorporated it into https://github.com/apache/arrow-datafusion/pull/7840

Thanks again @parkma99